In [58]:
import pandas as pd
from scipy import stats
import numpy as np

# The company wishes to clarify whether there is any increase in sales after stepping into digital marketing


In [59]:
data=pd.read_csv("C:\DSA\Sales_add.csv")

In [60]:
data.head()

Month      Region      Manager  Sales_before_digital_add(in $)  \
0  Month-1  Region - A  Manager - A                          132921   
1  Month-2  Region - A  Manager - C                          149559   
2  Month-3  Region - B  Manager - A                          146278   
3  Month-4  Region - B  Manager - B                          152167   
4  Month-5  Region - C  Manager - B                          159525   

   Sales_After_digital_add(in $)  
0                         270390  
1                         223334  
2                         244243  
3                         231808  
4                         258402

In [61]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22 entries, 0 to 21
Data columns (total 5 columns):
 #   Column                          Non-Null Count  Dtype 
---  ------                          --------------  ----- 
 0   Month                           22 non-null     object
 1   Region                          22 non-null     object
 2   Manager                         22 non-null     object
 3   Sales_before_digital_add(in $)  22 non-null     int64 
 4   Sales_After_digital_add(in $)   22 non-null     int64 
dtypes: int64(2), object(3)
memory usage: 1008.0+ bytes


In [62]:
data[['Sales_before_digital_add(in $)','Sales_After_digital_add(in $)']].describe()

Sales_before_digital_add(in $)  Sales_After_digital_add(in $)
count                       22.000000                      22.000000
mean                    149239.954545                  231123.727273
std                      14844.042921                   25556.777061
min                     130263.000000                  187305.000000
25%                     138087.750000                  214960.750000
50%                     147444.000000                  229986.500000
75%                     157627.500000                  250909.000000
max                     178939.000000                  276279.000000

In [63]:
# Hypothesis
# Reject H0: There is an increase in sales after stepping into digital marketing
# Reject Ha: There is no increase in sales after stepping into digital marketing 

In [64]:
Sales_Before_Digital=data[["Sales_before_digital_add(in $)"]]
Sales_After_Digital=data[["Sales_After_digital_add(in $)"]]

In [65]:
tval,pval = stats.ttest_rel(Sales_After_Digital,Sales_Before_Digital)

In [66]:
stats.ttest_rel(Sales_After_Digital,Sales_Before_Digital)

Ttest_relResult(statistic=array([12.09070525]), pvalue=array([6.336667e-11]))

In [67]:
tval

array([12.09070525])

In [68]:
tcriteria=stats.t.ppf((1-0.05), 21)

In [69]:
stats.t.ppf((1-0.05), 21)

1.7207429028118775

In [70]:
if tval>tcriteria:
    print(' Reject H0: there is an increase in sales after stepping into digital marketing')
else:
    print('Reject Ha: There is no increase in sales after stepping into digital marketing')

 Reject H0: there is an increase in sales after stepping into digital marketing


Based on the paired(related) t tset of the sales data,before and after stepping into digital marketing, it is observed that the t value(12.09) is higher than the  critical table value of  1.7204. So there is an enough evidence to conclude that an increase in sales after stepping into digital marketing.

# The company needs to check whether there is any dependency between the Features "Region" and "Manager"


In [71]:
mycrosstable=pd.crosstab(data.Region,data.Manager)

In [72]:
mycrosstable

Manager     Manager - A  Manager - B  Manager - C
Region                                           
Region - A            4            3            3
Region - B            4            1            2
Region - C            1            3            1

# Hypothesis
# Reject H0: There no dependancy between the features Region and Manager
# Reject Ha: There is a dependancy between the features Region and Manager


In [73]:
from scipy.stats import chi2_contingency

In [74]:
chiVal,pVal,df,exp=chi2_contingency(mycrosstable)

In [75]:
chiVal,pVal,df,exp

(3.050566893424036,
 0.5493991051158094,
 4,
 array([[4.09090909, 3.18181818, 2.72727273],
        [2.86363636, 2.22727273, 1.90909091],
        [2.04545455, 1.59090909, 1.36363636]]))

In [79]:
if pVal<0.05:
    print(' Reject H0: there is  daependancy between the feaures Region and Manager')
else:
    print('Reject Ha: There is no dependancy between the feaures Region and manager')

Reject Ha: There is no dependancy between the feaures Region and manager


Based on the chi2_squired test of the features between Region anad Manager , it is observed that the p value(0.549) is higher than the p significance value of 0.05 .It is failed to reject null hypothesis and to conclude that there is no  dependancy between features between Region and Manager